src/backend/core/api/viewsets.py

```diff
@@ -343,14 +343,13 @@ def get_queryset(self):
         queryset = queryset.filter(team=self.kwargs["team_id"])
 
         if self.action in {"list", "retrieve"}:
-            # Limit to team access instances related to a team THAT also has a team access
-            # instance for the logged-in user (we don't want to list only the team access
-            # instances pointing to the logged-in user)
+            # Determine which role the logged-in user has in the team
             user_role_query = models.TeamAccess.objects.filter(
-                team__accesses__user=self.request.user
+                user=self.request.user, team=self.kwargs["team_id"]
             ).values("role")[:1]
 
             queryset = (
+                # The logged-in user should be a member of a team to see its accesses
```

Comment:
you mean "part of a team" because "member" is an actual role...
```suggestion
                # The logged-in user should be part of a team to see its accesses
```